### PR TITLE
fix: make init_resource_with_retrier/4 only 'throw' when failed

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
The return value and failure pattern of init_resource_with_retrier/4 should be the same as init_resource/4.
The issue was introduced by this one: https://github.com/emqx/emqx/pull/9199

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/` dir
- [ ] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl [hunk 1]
```diff
@@ -675,16 +675,17 @@ init_resource(Module, OnCreate, ResId, Config) ->
...
-                                    status = #{is_alive => true}},
-        emqx_rule_registry:add_resource_params(ResParams)
-    catch Class:Reason:ST ->
-        emqx_rule_monitor:ensure_resource_retrier(ResId),
-        erlang:raise(Class, {init_resource, Reason}, ST)
-    end.
+    Params =
+        try
+            Module:OnCreate(ResId, Config)
+        catch Class:Reason:ST ->
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_engine.erl:681]: Do not log stacktrace when it’s a throw?

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl [hunk 2]
```diff
@@ -675,16 +675,17 @@ init_resource(Module, OnCreate, ResId, Config) ->
...
+            emqx_rule_monitor:ensure_resource_retrier(ResId),
+            throw({{Module, OnCreate}, {Class, Reason, ST}})
```
[the remark this answers is not in the document]
Comment (terry-xiaoyu) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl:683]: fixed

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl (hunk 1 above)
Comment (terry-xiaoyu) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl:681]: fixed

**Review by xiangfangyang-tech — APPROVED** [review]
tested good

**Pull request merged**